onto-viewer-core/src/main/java/org/edmcouncil/spec/ontoviewer/core/ontology/data/handler/InferableRestrictionHandler.java

```diff
@@ -194,11 +191,15 @@ private List<OwlAxiomPropertyValue> filterPropertiesWithRestrictions(Map<String,
         .collect(Collectors.toList());
   }
 
-  private List<PropertyValue> mergeProperties(Map<String, List<PropertyValue>> properties) {
-    List<PropertyValue> mergedProperties = new ArrayList<>();
+  private Set<PropertyValue> mergeProperties(Map<String, List<PropertyValue>> properties) {
+    Set<PropertyValue> mergedProperties = new HashSet<>();
     for (List<PropertyValue> values : properties.values()) {
-      mergedProperties.addAll(values);
+      for (PropertyValue value : values) {
+        if (!(properties.get("@viewer.function.anonymous_ancestor").contains(value))) {
```

Comment:
This line causes the following warning:

> ... WARN  169  o.e.s.o.c.o.d.h.classes.ClassHandler      : Unable to handle class http://example.com/test1/A. Details: Cannot invoke "java.util.List.contains(Object)" because the return value of "java.util.Map.get(Object)" is null
> java.lang.NullPointerException: Cannot invoke "java.util.List.contains(Object)" because the return value of "java.util.Map.get(Object)" is null

in the test:
https://github.com/edmcouncil/onto-viewer/blob/e7edd9e36b467422ab617dbd155984ef5a508b3d/onto-viewer-toolkit/src/test/java/org/edmcouncil/spec/ontoviewer/toolkit/handlers/OntologyTableDataExtractorTest.java#L76-L95